[GRID 5 & 6] Add grid default image and folder image

Koerty opened the pull request:
This is the step 5 and 6 of the grid roadmap. I decided to ship them both at the same time as they share very similar code.

This add default image and folder image to the grid. They can be configured by the theme with the defaultImage and folderImage properties of the imagegrid.

EDIT : I only tested this on my laptop, because I don't have my pie under the hand. It shouldn't be a problem, but if someone could test it on his pie and confirm it, that would be nice.

Koerty commented:
Idk who created the different images for ES (on/off buttons, arrows, ...) but maybe we should ask some designer on the forum to do the same for the default rom image and the folder image.

First point, is legally speaking, idk who created them originally, I just took them from Jacobfk20's grid view.

Second point, is the default rom image look ok, but the folder image don't look consistent with other ES UI elements. For example, most of them have a dark grey border around them (on/off buttons, checkbox, ...) while the folder image have none.

jrassa commented on es-core/src/components/ImageGridComponent.h:
```diff
@@ -96,7 +101,21 @@ void ImageGridComponent<T>::add(const std::string& name, const std::string& imag
+	// so we don't want to bring that dependence to FileData here
+	if (obj->getType() == 2)
+	{
+		entry.data.texture = mFolderTexture;
```
Folders can have images specified in gamelist.xml  We should probably use that if present vs. defaulting to folder image.

jrassa commented:
> Idk who created the different images for ES but maybe we should ask some designer on the forum to do the same for the default rom image and the folder image.
> 
> First point, is legally speaking, idk who created them originally, I just took them from Jacobfk20's grid view.
> 
> Second point, is the default rom image look ok, but the folder image don't look consistent with other ES UI elements. For example, most of them have a dark grey border around them (off button, checkboxs, ...) while the folder image have none.

Its probably not a big deal since most themes will probably swap them out, but feel free to ask.  I'm sure someone will step up.

Koerty commented:
I didn't know folder could have metadata aswell. I swapped the 2 "if" conditions to handle this case.

In that case we should rename the 2 theming elements names, for either : 
`defaultRomImage`
`defaultFolderImage`
or 
`romImage`
`folderImage`

As it make more sense than the current syntax since folder can have custom images : 
`defaultImage`
`folderImage`

Koerty commented:
> Its probably not a big deal since most themes will probably swap them out, but feel free to ask. I'm sure someone will step up.

Yeah, let's not bother with that.

I changed `defaultImage` to `gameImage`, because the old name `defaultImage` implied that `folderImage` would always be used (even if the gamelist define custom folder image), and `defaultGameImage` + `defaultFolderImage` were too long and not consistent with other elements with similar name.

I tested and rebased it. Ready when you are.

jrassa commented:
Thanks.
